### DISCUSSION TITLE
Lesson - 7 why in testOwnerIsMsgSender, the i_owner is msg.sender ?

### DISCUSSION THREAD

**Discussion by getzelus** [Cyfrin/foundry-full-course-cu#639] [General]
In the second video at 41:00, I wonder why the result of the test is good for this function : 
```
  function testOwnerIsMsgSender() public{
        assertEq(fundMe.i_owner(), msg.sender);
    }
```

Because the FundMeTest contract creates DeployFundMe.
Then DeployFundMe creates FundMe.

**So the i_owner of fundMe should be the address of deployFundMe** and not us (msg.sender) ?

**Comment by Struka9** [comment]
From Foundry's startBroadcast docs:
`Using the address that calls the test contract or the address / private key provided as the sender, has all subsequent calls (at this call depth only and excluding cheatcode calls) create transactions that can later be signed and sent onchain.`

So from what I understand using calling startBroadcast from tests will set up the address to be the same as the one calling the test functions, that's why it makes sense for msg.sender to be the same as i_owner.

edit: vm.startBroadcast is used in the Deploy script.

**Comment by shikhar229169** [comment]
Check out this:
https://github.com/Cyfrin/foundry-full-course-f23/discussions/504
https://github.com/Cyfrin/foundry-full-course-f23/discussions/513
